subprojects/core/src/main/java/org/gradle/plugin/use/internal/DefaultPluginRequest.java

```diff
@@ -57,6 +57,17 @@ public String getVersion() {
         return version;
     }
 
+    @Nullable
+    @Override
+    public String getArtifact() {
+        return null;
+    }
+
+    @Override
+    public boolean isConfigured() {
```

Comment:
I wouldn't do that, as it is kind of half-baked. If you really wanted tracing through resolution rules, you'd need to record every mutation. We don't even do that for normal resolution rules yet.

What we could do (but what I'd leave out of this PR since it is big enough) is passing *both* the requested and the target selector into the resolver chain, so they could do better reporting.